### PR TITLE
Cluster Local Gateway for Istio 1.3.1 - for KFServing

### PR DESCRIPTION
**Pull request by krishnadurai**
**Which issue is resolved by this Pull Request:**
Parially addresses kubeflow/kfserving#652

**Description of your changes:**

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate-changed-only`
    3. `make test`

/cc @yuzisun @animeshsingh

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/875)

### CODE REVIEW

**Comment by krishnadurai** [comment]
/hold

Needs testing on Kubeflow deployment

File: istio-1-3-1/cluster-local-gateway-1-3-1/base/cluster-role-binding.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+kind: ClusterRoleBinding
+metadata:
+  name: istio-multi
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: istio-reader
+subjects:
+- kind: ServiceAccount
+  name: istio-multi
```
Comment (yuzisun) [inline istio-1-3-1/cluster-local-gateway-1-3-1/base/cluster-role-binding.yaml]: nit: can we fix the new lines? there are other places.

File: istio-1-3-1/cluster-local-gateway-1-3-1/base/pod-disruption-budget.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
```
Comment (yuzisun) [inline istio-1-3-1/cluster-local-gateway-1-3-1/base/pod-disruption-budget.yaml]: do we need the hpa also?  looks we have it for ingress gateway

**Comment by animeshsingh** [comment]
@Tomcli tested on kubeflow cluster

File: istio-1-3-1/cluster-local-gateway-1-3-1/base/cluster-role-binding.yaml (hunk 1 above)
Comment (krishnadurai) [reply istio-1-3-1/cluster-local-gateway-1-3-1/base/cluster-role-binding.yaml]: Sure

File: istio-1-3-1/cluster-local-gateway-1-3-1/base/pod-disruption-budget.yaml (hunk 2 above)
Comment (krishnadurai) [reply istio-1-3-1/cluster-local-gateway-1-3-1/base/pod-disruption-budget.yaml]: @yuzisun Interesting thought. The default Helm charts for Istio do not come with one. Should we add one like this?

```
apiVersion: autoscaling/v2beta1
kind: HorizontalPodAutoscaler
metadata:
  labels:
    app: cluster-local-gateway
    istio: cluster-local-gateway
  name: cluster-local-gateway
spec:
  maxReplicas: 5
  metrics:
  - resource:
      name: cpu
      targetAverageUtilization: 80
    type: Resource
  minReplicas: 1
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: cluster-local-gateway
```

Comment (yuzisun) [reply istio-1-3-1/cluster-local-gateway-1-3-1/base/pod-disruption-budget.yaml]: sgtm

**Comment by krishnadurai** [comment]
Tested this with Istio 1.3.1 with kfctl_istio_dex. Even when the cluster-local-gateway seems to function well, we seem to have an issue setting up the ingress:

```
kubectl get ksvc,configuration,revision,route
NAME                                                           URL                                                           LATESTCREATED                            LATESTREADY                              READY     REASON
service.serving.knative.dev/flowers-sample-predictor-default   http://flowers-sample-predictor-default.default.example.com   flowers-sample-predictor-default-wz9jr   flowers-sample-predictor-default-wz9jr   Unknown   IngressNotConfigured
NAME                                                                 LATESTCREATED                            LATESTREADY                              READY   REASON
configuration.serving.knative.dev/flowers-sample-predictor-default   flowers-sample-predictor-default-wz9jr   flowers-sample-predictor-default-wz9jr   True
NAME                                                                  CONFIG NAME                        K8S SERVICE NAME                         GENERATION   READY   REASON
revision.serving.knative.dev/flowers-sample-predictor-default-wz9jr   flowers-sample-predictor-default   flowers-sample-predictor-default-wz9jr   1            True
NAME                                                         URL                                                           READY     REASON
route.serving.knative.dev/flowers-sample-predictor-default   http://flowers-sample-predictor-default.default.example.com   Unknown   IngressNotConfigured
```

Whereas upgrading Istio configuration to 1.4.2 seems to solve this issue.

I'll file a separate issue for this problem, as I've isolated this to our Istio setup.

@Tomcli could you please describe your test setup.

@yuzisun as discussed, we could look to independenly merge this change.

**Pull request merged**